apps/studio/electron/preload/webview/dom.ts

```diff
@@ -1,10 +1,62 @@
 import { ipcRenderer } from 'electron';
+import { EditorAttributes } from '@onlook/models/constants';
 import { getOrAssignUuid } from './elements/helpers';
 import { WebviewChannels } from '@onlook/models/constants';
 import { getUniqueSelector, isValidHtmlElement } from '/common/helpers';
 import type { LayerNode } from '@onlook/models/element';
 
+function markDynamicElements(root: HTMLElement) {
+    const containers = root.querySelectorAll(`[${EditorAttributes.DATA_ONLOOK_ID}]`);
+
+    containers.forEach((container) => {
+        const children = Array.from(container.children);
+        if (children.length < 2) {
+            return;
+        }
+
+        const groups = children.reduce(
+            (acc, child) => {
+                if (!isValidHtmlElement(child)) {
+                    return acc;
+                }
+
+                const signature = [
+                    child.tagName,
+                    child.className,
+                    child.getAttribute(EditorAttributes.DATA_ONLOOK_ID),
```

Comment:
My understanding: We're making an assumption that if an element has the same `data-onlook-id` as its sibling, it would be part of an array? 

One edge case is if an element is an instance and it's used twice, these would have the same signature?

```tsx
export default function Page() {
    return (
        <div>
            <Instance />
            <Instance />
        </div>
    );
}

function Instance() {
    return <div>Instance</div>;
}
```

Reply:
Thought this is really close! I wonder if there's something we can do on the plugin side to populate these attributes